src/models/shelf/spec.ts

```diff
@@ -27,6 +25,8 @@ export interface ShelfUnitSpec {
   anyEncodings: ShelfAnyEncodingDef[];
 
   config: Config;
+
+  filters?: Transform[];
```

Comment:
> Should filters be an optional parameter? or should it be an empty array by default?

I don't have strong opinion about this, but I guess empty array would lead to less code (no need to check if something is defined.)